plugins/modules/netbox_device_interface.py

```diff
@@ -298,9 +290,6 @@ def main():
                 options=dict(
                     device=dict(required=False, type="raw"),
                     name=dict(required=True, type="str"),
-                    form_factor=dict(
```

Comment:
Removing options requires a new major release, which I would really avoid for now until you have a new minor (or patch) release that fixes the current issues. A new major release has to wait for Ansible 5 and cannot be included in the next Ansible 4 release (which will happen ~next Tuesday BTW).

Reply:
Thanks @felixfontein , this is great feedback! I did try to add the `'removed_from_collection': 'netbox.netbox'` but then the tests said `ERROR: plugins/modules/netbox_device_interface.py:0:0: collection-deprecated-version: Argument 'form_factor' in argument_spec found in data has a deprecated removed_in_version '0.3.0', i.e. the version is less than or equal to the current version of this collection (3.1.1)`.

Does this mean it should really have been removed before? And should I now just bump the `removed_in_version` to 4.0.0 to resolve this as smooth as possible? And then I guess the tests will prompt me if I'm not removing it from version 4.0.0?

Reply:
Yes, it should have been removed in 1.0.0. (Or maybe the value '0.3.0' is also wrong, maybe it was the version when the deprecation was added?)

In any case, you could either add an ignore.txt entry for this, or replace 0.3.0 by 4.0.0, as you suggested. In the latter case, you (or whoever else) will be reminded once you change the version to 4.0.0 and run sanity checks.

Reply:
This is super useful feedback!  Thanks @felixfontein !